Sources/RedCat/ObservableStore.swift

```diff
@@ -119,16 +137,19 @@ final class ConcreteStore<Reducer : ErasedReducer> : ObservableStore<Reducer.Sta
             // the action must have been enqueued by the below while loop
             return
         }
-        
-        observers.notifyAll()
-        
-        dispatchActions()
-        
+			
+				let oldState = _state
+				let newState = dispatchActions()
+			
+				guard !isEqual(oldState, newState) else { return }
+				objectWillChange.send()
+				_state = newState
+				observers.notifyAll(old: oldState, new: newState, action: action)
     }
     
     @usableFromInline
-    func dispatchActions() {
-        
+		func dispatchActions() -> Reducer.State {
+        var newState = _state
```

Comment:
The main problem here is unnecessary subscribers updates
Using something like `.removeDuplicates()` in the subscribers themselves means storing a copy of the state in the subscribers, which we would like to avoid
Store could call the notify of subscribers in the `didSet` of `_state` where there are both old and new values

Reply:
A lot of my other comments point to the same issue. I think the best place to discuss this is in ObservableStore.swift.